### PR TITLE
Fixing my previous mistakes introduced with my pull request

### PR DESCRIPTION
**Pull request by x0f5c3**
Fixed my mistake cause by cloning in a directory that doesn't exist, created a new function to clone the full repository using an absolute path. Added prompt theme chosen by checking if user's terminal supports colors.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -96,40 +96,51 @@ impl Config {
...
+                .paged(true)
+                .clear(true)
+                .interact_on(&term)?;
             if chosen_in == 0 {
-                return super::git::clone_single_branch(
-                    &self.repo_directory,
-                    &self.repo.url,
-                    &self.repo.branch,
-                );
+                return crate::git::clone_full(&self.repo_directory, &self.repo.url);
```
Comment (tversteeg) [inline src/config.rs:136]: Can you `use crate::git` at the top and then use `git::..` here?

File: src/git.rs [hunk 2]
```diff
@@ -91,6 +91,12 @@ pub fn pull<P: AsRef<Path>>(dir: &P, branch: &str) -> Result<bool> {
     )
     .context("failed pulling in git: merge")
 }
+/// Clone a full repository
```
Comment (tversteeg) [inline src/git.rs:94]:
```suggestion

/// Clone a full repository.
```

File: src/git.rs [hunk 3]
```diff
@@ -91,6 +91,12 @@ pub fn pull<P: AsRef<Path>>(dir: &P, branch: &str) -> Result<bool> {
...
+pub fn clone_full(dir: &str, url: &str) -> Result<bool> {
+    // Create a dummy path since we specify absolute path for cloning
+    let dummy_path = std::path::PathBuf::from("./");
```
Comment (tversteeg) [inline src/git.rs:97]: Can you `use std::path::PathBuf` at the top and then use `PathBuf::from("./");` here?

File: src/config.rs (hunk 1 above)
Comment (x0f5c3) [reply src/config.rs]: Done in new commit

**Comment by x0f5c3** [comment]
I'm sorry for making a little bit of a mess out of this.

File: src/git.rs [hunk 4]
```diff
@@ -92,6 +92,13 @@ pub fn pull<P: AsRef<Path>>(dir: &P, branch: &str) -> Result<bool> {
     .context("failed pulling in git: merge")
 }
 
+/// Clone a full repository
```
Comment (tversteeg) [inline src/git.rs:95]:
```suggestion
/// Clone a full repository.
```

File: src/config.rs [hunk 5]
```diff
@@ -96,40 +97,51 @@ impl Config {
...
-                    &self.repo.url,
-                    &self.repo.branch,
-                );
+                return git::clone_full(&self.repo_directory, &self.repo.url);
             } else {
                 fs::DirBuilder::new()
                     .recursive(true)
                     .create(&self.repo_directory)?;
-                return super::git::set_remote(&self.repo_directory, &self.repo.url);
+                // I obviously forgot that the repo needs to be initialized
```
Comment (tversteeg) [inline src/config.rs:142]:
```suggestion
```

File: src/config.rs [hunk 6]
```diff
@@ -96,40 +97,51 @@ impl Config {
 
     /// Ask the user if he wants to change the repo path, clone it or create locally, or abort.
```
Comment (tversteeg) [inline src/config.rs:98]:
```suggestion
    /// Ask the user if they want to change the repo path, clone it or create locally, or abort.
```

**Comment by tversteeg** [comment]
Thanks!

**Pull request merged**